### DISCUSSION TITLE
Negativ values again :)

### DISCUSSION THREAD

**Discussion by jojogib** [RomanKubiak/ctrlr#533] [Q&A]
Hey everyone  
local bi = BigInteger(value)

  local MS = bi:getBitRangeAsInt(0,4)
  local LS = bi:getBitRangeAsInt(4, 4)
  local MS1 = bi:getBitRangeAsInt(8,4)
  local LS1 = bi:getBitRangeAsInt(12,4)

this works for me with positive values but i need to go negative 
starting by:
0 = 00 00 00 00
-1 = 0f 0f 0f 0f 
-2 = 0e 0f 0f 0f
-3 = 0d 0f 0f 0f
...
-9999 = 01 0f 08 0d
anyone know how to calculate`?
Thank you

**Comment by damiensellier** [comment]
The range is always -9999 to +9999 ?

**Reply by jojogib** [reply]
yes

**Comment by dnaldoog** [comment]
Here is how I would do it! 
<pre>
ushort= 65535 --  mask signed short integer
myMethod = function(--[[ CtrlrModulator --]] mod --[[ number --]], value --[[ number --]], source)
    local bi = BigInteger(luaIntMask(value))
    local MS = bi:getBitRangeAsInt(0, 4)
    local LS = bi:getBitRangeAsInt(4, 4)
    local MS1 = bi:getBitRangeAsInt(8, 4)
    local LS1 = bi:getBitRangeAsInt(12, 4)
    console(string.format("%.2x %.2x %.2x %.2x", MS, LS, MS1, LS1))
end
function luaIntMask(val, mask)
    local mask = mask or ushort
    return bit.band(val, mask)
end
</pre>

[signed number 4 bit nibbleized_2_0.zip](https://github.com/RomanKubiak/ctrlr/files/10496972/signed.number.4.bit.nibbleized_2_0.zip)

![masknibbleizenegativepanel](https://user-images.githubusercontent.com/76831683/214463561-93314d9a-0e3d-4233-aacf-302c46bbab50.png)

**Reply by jojogib** [reply]
awesome it worked. I really need to study the lua basics. I have 0 programming experience so I dont know where to start.
